### DISCUSSION TITLE
Container provider configuration

### DISCUSSION THREAD

**Discussion by mdelapenya** [testcontainers/testcontainers-go#504] [Ideas]
At this moment, it's possible to set the provider at the `ContainerRequest` level, but I'm not sure if somebody want to select the provider for each container.

Instead, it seems to me that an initial configuration, probably at the properties level would be better.

Thoughts?

**Comment by prskr** [comment]
I guess in 99% of the cases the provider will **always** be Docker even for Podman environments because the provider basically only changes the name of the default network from _bridge_ to _podman_ which only matters n cases when (multiple) networks are specified in the `ContainerRequest`.

I'd be interested how often this feature is actually used because I'd somehow expect that users that want to use multiple containers/networks would rather go with a `docker-compose` approach as it is not really convenient to configure everything in Go and a `docker-compose` config would have the advantage that you can also use it for other scenarios as well.

If testcontainers would support to configure the default provider from a `.testcontainers.properties` then I think it'd also make sense to support some kinda `TC_CONTAINER_PROVIDER` environment variable to allow dynamic overrides if necessary e.g. in CI environments.

Personally I can't recall to have ever used container networks at all with testcontainers :smile: but maybe I'm just not doing 'complex enough' tests :smile:

**Reply by mdelapenya** [reply]
>If testcontainers would support to configure the default provider from a .testcontainers.properties then I think it'd also make sense to support some kinda TC_CONTAINER_PROVIDER environment variable to allow dynamic overrides if necessary e.g. in CI environments.

Indeed, that would be needed. Will check internally with the java folks if they support env-vars as properties

**Comment by mdelapenya** [comment]
Closing this discussion as outdated, the discovery mechanism should detect the host and socket, which would be enough.
